src/db/entity/User.ts

```diff
@@ -0,0 +1,29 @@
+import {
+    Entity, PrimaryGeneratedColumn, Column, OneToMany, JoinColumn, BaseEntity
+} from "typeorm"
+import { Search } from "./Search"
+
+@Entity()
+export class User extends BaseEntity {
+    @Column()
+    username: string;
+
+    @Column()
+    lang: string;
+
+    @PrimaryGeneratedColumn("uuid")
+    token: string;
+
+
+    @OneToMany(() => Search, (search) => search.user, {
+        cascade: true
+    }) // note: we will create user property in the search class below
+    searches: Search[];
+
+    addSearch(search: Search) {
```

Comment:
You added this function on the User class, but you also have `queries.ts`. I think both solution are ok but I would stick to having 1 solution, otherwise in a large codebase it would be hard to find what happens where

Reply:
the function in queries.ts calls the addSearch func in the user entity, its not ok?